easybuild/tools/modules.py

```diff
@@ -209,8 +209,11 @@ def modules(self):
 
     def set_and_check_version(self):
         """Get the module version, and check any requirements"""
+        lmod_depends_on_min = '7.6.1'
         if self.COMMAND in MODULE_VERSION_CACHE:
             self.version = MODULE_VERSION_CACHE[self.COMMAND]
+            self.has_depends_on = (isinstance(self, Lmod) and
```

Comment:
I'd go with `supports_depends_on` rather than `has_depends_on`

`supports_depends_on` should be set to `False` in `ModulesTool.__init__`

and then in `Lmod.__init__` it should be set based on the Lmod version after the `super` call (don't use `isinstance(self, Lmod)`)